worlds/soe/__init__.py

```diff
@@ -0,0 +1,250 @@
+
+from . import Logic  # load logic mixin
+from .Options import soe_options
+from .Patch import generate_patch
+
+"""
+In evermizer:
+
+Items are uniquely defined by a pair of (type, id).
+For most items this is their vanilla location (i.e. CHECK_GOURD, number).
+
+Items have `provides`, which give the actual progression
+instead of providing multiple events per item, we iterate through them in Logic.py
+    e.g. Found any weapon
+
+Locations have `requires` and `provides`.
+Requirements have to be converted to (access) rules for AP
+    e.g. Chest locked behind having a weapon
+Provides could be events, but instead we iterate through the entire logic in Logic.py
+    e.g. NPC available after fighting a Boss
+
+Rules are special locations that don't have a physical location
+instead of implementing virtual locations and virtual items, we simply use them  in Logic.py
+    e.g. 2DEs+Wheel+Gauge = Rocket
+
+Rules and Locations live on the same logic tree returned by pyevermizer.get_logic()
+
+TODO: for balancing we may want to generate Regions (with Entrances) for some
+common rules, place the locations in those Regions and shorten the rules.
+"""
+
+_id_base = 64000
+_id_offset: typing.Dict[int, int] = {
+    pyevermizer.CHECK_ALCHEMY: _id_base + 0,  # alchemy 64000..64049
+    pyevermizer.CHECK_BOSS: _id_base + 50,  # bosses 64050..6499
+    pyevermizer.CHECK_GOURD: _id_base + 100,  # gourds 64100..64399
+    pyevermizer.CHECK_NPC: _id_base + 400,  # npc 64400..64499
+    # TODO: sniff 64500..64799
+}
+
+# cache native evermizer items and locations
+_items = pyevermizer.get_items()
+_locations = pyevermizer.get_locations()
+# fix up texts for AP
+for _loc in _locations:
+    if _loc.type == pyevermizer.CHECK_GOURD:
+        _loc.name = f'{_loc.name} #{_loc.index}'
+
+
+def _get_location_mapping() -> typing.Tuple[typing.Dict[str, int], typing.Dict[int, pyevermizer.Location]]:
+    name_to_id = {}
+    id_to_raw = {}
+    for loc in _locations:
+        apid = _id_offset[loc.type] + loc.index
+        id_to_raw[apid] = loc
+        name_to_id[loc.name] = apid
+    name_to_id['Done'] = None
+    return name_to_id, id_to_raw
+
+
+def _get_item_mapping() -> typing.Tuple[typing.Dict[str, int], typing.Dict[int, pyevermizer.Item]]:
+    name_to_id = {}
+    id_to_raw = {}
+    for item in _items:
+        if item.name in name_to_id:
+            continue
+        apid = _id_offset[item.type] + item.index
+        id_to_raw[apid] = item
+        name_to_id[item.name] = apid
+    name_to_id['Victory'] = None
+    return name_to_id, id_to_raw
+
+
+class SoEWorld(World):
+    """
+    Secret of Evermore is a SNES action RPG. You learn alchemy spells, fight bosses and gather rocket parts to visit a
+    space station where the final boss must be defeated. 
+    """
+    game: str = "Secret of Evermore"
+    options = soe_options
+    topology_present: bool = True
+    remote_items: bool = False  # True only for testing
+    data_version = 0
```

Comment:
might want to bump that to 1 for the PR, though I don't mind doing that post-merge myself.